Source/SpatialGDK/Private/SpatialInterop.cpp

```diff
@@ -613,30 +613,38 @@ void USpatialInterop::ResolvePendingIncomingObjectUpdates(UObject* Object, const
 		USpatialActorChannel* DependentChannel = ChannelProperties.Key;
 		FPendingIncomingProperties& Properties = ChannelProperties.Value;
 
-		// Trigger pending updates.
-		DependentChannel->PreReceiveSpatialUpdate(Object);
-
+		// Replicated Properties
 		TSet<UProperty*> RepNotifies;
-		for (const FRepHandleData* RepData : Properties.Key)
+		for (auto& ObjsToReplicate : Properties.Key)  // TPair<UObject*, TArray<const FRepHandleData*>>
 		{
-			ApplyIncomingReplicatedPropertyUpdate(*RepData, DependentChannel->Actor, &Object, RepNotifies);
-			UE_LOG(LogSpatialGDKInterop, Log, TEXT("%s: Received queued object replicated property update. actor %s (%lld), property %s"),
-				*SpatialOSInstance->GetWorkerId(),
-				*DependentChannel->Actor->GetName(),
-				DependentChannel->GetEntityId().ToSpatialEntityId(),
-				*RepData->Property->GetName());
+			// Iterate through our list of target objects, either an actor or component
+			// For each, we resolve all replicated properties under that object
+			UObject* TargetObject = ObjsToReplicate.Key;
+			DependentChannel->PreReceiveSpatialUpdate(TargetObject);
+
+			for (const FRepHandleData* RepData : ObjsToReplicate.Value) {  // Iterating through TArray<const FRepHandleData*>
+				ApplyIncomingReplicatedPropertyUpdate(*RepData, TargetObject, &IncomingObject, RepNotifies);
+				UE_LOG(LogSpatialGDKInterop, Log, TEXT("%s: Received queued object replicated property update. Channel actor (root actor) %s (%lld), target object %s, property %s"),
```

Comment:
I don't think "root actor" is a thing. I feel like "Channel actor" is descriptive enough.

Reply:
done